### PR TITLE
MINOR: Remove ReplicaManagerTest.initializeLogAndTopicId

### PR DESCRIPTION
**Pull request by ijuma**
The workaround is not required with mockito.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala [hunk 1]
```diff
@@ -1953,6 +1935,7 @@ class ReplicaManagerTest {
     val mockLogMgr: LogManager = mock(classOf[LogManager])
     when(mockLogMgr.liveLogDirs).thenReturn(config.logDirs.map(new File(_).getAbsoluteFile))
     when(mockLogMgr.getOrCreateLog(ArgumentMatchers.eq(topicPartitionObj), ArgumentMatchers.eq(false), ArgumentMatchers.eq(false), any())).thenReturn(mockLog)
+//    when(mockLogMgr.getLog(topicPartitionObj, isFuture = false)).thenReturn(Some(mockMergedLog))
```
Comment (divijvaidya) [inline core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala]: is this intended?

Comment (Kvicii) [reply core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala]: I think if this unused, we should remove it ?

Comment (ijuma) [reply core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala]: No :) The test doesn't pass without this line. Fixed.

**Review by Kvicii — APPROVED** [review]

**Comment by divijvaidya** [comment]
The failing test `testSnapshotOnlyAfterConfiguredMinBytes` is failing due to a bug which creates a snapshot with LONG_MAX epoch. This is documented in https://issues.apache.org/jira/browse/KAFKA-13943. The bug has been fixed in https://github.com/apache/kafka/pull/12224. @ijuma perhaps you may be interested to review that one after this? It will help us reduce flaky test behaviour.

**Review by divijvaidya — APPROVED** [review]
LGTM!

**Review by jsancio — APPROVED** [review]
LGTM. Thanks for the changes @ijuma

**Pull request merged**